src/v3/src/components/InputPassword/InputPassword.tsx

```diff
@@ -81,9 +83,22 @@ const InputPassword: UISchemaElementComponent<UISchemaElementComponentWithValida
     <Box>
       <InputLabel
         htmlFor={name}
-        required={required}
+        // To prevent asterisk from shifting far right
+        sx={{ justifyContent: showAsterisk ? 'flex-start' : undefined }}
       >
         {label}
+        {showAsterisk && (
+          <Box
+            component="span"
+            className="no-translate"
+            aria-hidden
+          >
+            &nbsp;&#42;
+          </Box>
+        )}
+        {required === false && (
```

Comment:
are both 'showAsterisk` and 'required' properties needed?  Do they both share the same purpose of saying if the input is required or not?

Reply:
Or is it because maybe the case where we want an input to be not required, but still show an asterisk, or vice versa?

Reply:
Ah yes, so the asterisk will only show up on the profile enrollment page (per UX's request) and since we cant override the required property (because we need to know for those fields that are not required they should show the optional label) I had to create a new property to depict when we should show the asterisk. 

Added a blurb here: https://github.com/okta/okta-signin-widget/pull/3205/files#diff-bc22796207c263f6ef1317c9b0844cb530619b0f5ca1512fdb0314df25bd85faR245-R248

At the component level, we don't have the context of whether its profile enrollment or some other remediation, this is why I had to create a new property specifically for this.

Reply:
ahh okay cool I got it now, thanks for the explanation 🙏